### PR TITLE
feat(block-editor): add import from url feature

### LINKED ISSUE #29874: File field: "Import from URL" functionality
## Parent Issue
#29691 

## Task
Implement the "Import from URL" feature for **File Field**, which opens a popup for users to enter a URL and import a file from that location.

## Proposed Objective
User Experience

## Proposed Priority
Same as Parent Issue

## Acceptance Criteria
1. An "Import from URL" option is available in the File field
2. Clicking the option opens a popup with a URL input field
3. The system attempts to download and import the file from the provided URL
4. Error handling if the URL is invalid or the file cannot be imported
5. The popup remains open and displays an error message if the import fails
6. Successfully imported files are treated as if they were uploaded directly

## Assumptions & Initiation Needs
- Backend API for importing files from URLs is available
- Security measures for URL validation and file scanning are in place

### PR DESCRIPTION
**Pull request by nicobytes**
### Parent Issue

#29874 

### Proposed Changes
* Create `dot-form-import-url` component
* Create `FormImportUrlStore` store using signals
* Implement the "Import from URL" for Image and File fields

### Checklist
- [x] Tests
- [x] Translations
- [x] Security Implications Contemplated (add notes if applicable)

### Screenshots

https://github.com/user-attachments/assets/5362fd77-2d5e-4062-b0af-c8e89f79e271

This PR fixes: #29874

### CODE REVIEW

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/dot-edit-content-file-field.component.ts [hunk 1]
```diff
@@ -186,9 +198,56 @@ export class DotEditContentFileFieldComponent implements ControlValueAccessor, O
...
+            maskStyleClass: 'p-dialog-mask-transparent',
+            modal: true,
+            resizable: false,
+            position: 'center',
+            data: {
+                inputType: this.$field().fieldType
+            }
+        });
+
+        this.#dialogRef.onClose.pipe(filter((file) => !!file)).subscribe((file) => {
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/dot-edit-content-file-field.component.ts:236]: this suscribe not neet unsuscribe?

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/store/file-field.store.ts [hunk 2]
```diff
@@ -132,6 +125,13 @@ export const FileFieldStore = signalStore(
                     dropZoneActive: state
                 });
             },
+            setPreviewFile: (file: UploadedFile) => {
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/store/file-field.store.ts:128]: add doc.

**Review by oidacra — APPROVED** [review]

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/store/file-field.store.ts (hunk 2 above)
Comment (nicobytes) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/store/file-field.store.ts:128]: Fixed!

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/dot-edit-content-file-field.component.ts (hunk 1 above)
Comment (nicobytes) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/dot-edit-content-file-field.component.ts]: Fixed!

**Review by hmoreras — APPROVED** [review]

**Pull request merged**